From a69014a1b4e5b8068630abe3109f31eb64b6a076 Mon Sep 17 00:00:00 2001
From: Giulio Benetti <giulio.benetti@benettiengineering.com>
Date: Wed, 10 Aug 2022 00:10:51 +0200
Subject: [PATCH] nfsrahead: fix linking while static linking

-lmount must preceed -lblkid and to obtain this let's add:
`pkg-config --libs mount`
in place of:
`-lmount`
This ways the library order will always be correct.

Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
---
Pending Upstream:
https://patchwork.kernel.org/project/linux-nfs/patch/20220809223308.1421081-1-giulio.benetti@benettiengineering.com/
---
 tools/nfsrahead/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/nfsrahead/Makefile.am b/tools/nfsrahead/Makefile.am
index 845ea0d5..280a2eb4 100644
--- a/tools/nfsrahead/Makefile.am
+++ b/tools/nfsrahead/Makefile.am
@@ -1,6 +1,6 @@
 libexec_PROGRAMS = nfsrahead
 nfsrahead_SOURCES = main.c
-nfsrahead_LDFLAGS= -lmount
+nfsrahead_LDFLAGS= `pkg-config --libs mount`
 nfsrahead_LDADD = ../../support/nfs/libnfsconf.la
 
 man5_MANS = nfsrahead.man
-- 
2.34.1

